src/main/java/lottery/domain/LotteryNumber.java

```diff
@@ -0,0 +1,29 @@
+package lottery.domain;
+
+public class LotteryNumber implements Comparable<LotteryNumber> {
+    public final static int MIN =  1;
+    public final static int MAX = 45;
+
+    private final int number;
+
+    private static void sanitizeNumber(int number) {
+        if (number < MIN || number > MAX) {
+            throw new InvalidLotteryNumberException();
+        }
+    }
+
+    public LotteryNumber(int number) {
+        sanitizeNumber(number);
+        this.number = number;
+    }
```

Comment:
```suggestion
    private LotteryNumber(int number) {
        this.number = number;
    }
    
    public static LotteryNumber valueOf(int number) {
        sanitizeNumber(number);
        return new LotteryNumber(number); 
    }
```
보통 원시값 포장 객체는 이렇게 valueOf를 이용하여 만들거나 가져옵니다.

그리고 이 때
```java
LotteryNumber number1 = LotteryNumber.valueOf(1);
LotteryNumber number2 = LotteryNumber.valueOf(1);
System.out.println(number1 == number2); // true
```
이렇게 주소값을 통해서 비교했을 때(equals가 아닌) 똑같은 값이 나오게 하려면 어떻게 해야할까요?